Mathlib/Algebra/Order/Monoid/Unbundled/Pow.lean

```diff
@@ -58,24 +58,28 @@ end Left
 
 section Left
 
-variable [MulLeftMono M]
+variable [MulLeftMono M] {a : M} {n : ℕ}
 
 @[to_additive nsmul_left_monotone]
-theorem pow_right_monotone {a : M} (ha : 1 ≤ a) : Monotone fun n : ℕ ↦ a ^ n :=
+theorem pow_right_monotone (ha : 1 ≤ a) : Monotone fun n : ℕ ↦ a ^ n :=
   monotone_nat_of_le_succ fun n ↦ by rw [pow_succ]; exact le_mul_of_one_le_right' ha
 
 @[to_additive (attr := gcongr) nsmul_le_nsmul_left]
-theorem pow_le_pow_right' {a : M} {n m : ℕ} (ha : 1 ≤ a) (h : n ≤ m) : a ^ n ≤ a ^ m :=
+theorem pow_le_pow_right' {n m : ℕ} (ha : 1 ≤ a) (h : n ≤ m) : a ^ n ≤ a ^ m :=
   pow_right_monotone ha h
 
 @[to_additive nsmul_le_nsmul_left_of_nonpos]
-theorem pow_le_pow_right_of_le_one' {a : M} {n m : ℕ} (ha : a ≤ 1) (h : n ≤ m) : a ^ m ≤ a ^ n :=
+theorem pow_le_pow_right_of_le_one' {n m : ℕ} (ha : a ≤ 1) (h : n ≤ m) : a ^ m ≤ a ^ n :=
   pow_le_pow_right' (M := Mᵒᵈ) ha h
 
 @[to_additive nsmul_pos]
-theorem one_lt_pow' {a : M} (ha : 1 < a) {k : ℕ} (hk : k ≠ 0) : 1 < a ^ k :=
+theorem one_lt_pow' (ha : 1 < a) {k : ℕ} (hk : k ≠ 0) : 1 < a ^ k :=
   pow_lt_one' (M := Mᵒᵈ) ha hk
 
+@[to_additive]
+lemma le_self_pow (ha : 1 ≤ a) (hn : n ≠ 0) : a ≤ a ^ n := by
+  simpa using pow_le_pow_right' ha (Nat.one_le_iff_ne_zero.2 hn)
+
```

Comment:
What's the reason you're putting this lemma back again? Did you deprecate it by accident?

Reply:
It's a different lemma! The old one was about `GroupWithZero`